### DISCUSSION TITLE
Error querying dynamic routes

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#16272] [Help]
Hello community I just started with dynamic routes in my project, I have read the documentation but I have not managed to get the parameters I need from the route.

`Nextjs mentions this:`

For example, the route /post/abc will have the following query object:
```javascript
{ "pid": "abc" }
```
Similarly, the route /post/abc?foo=bar will have the following query object:

```javascript
{ "foo": "bar", "pid": "abc" }
```
`And my code is this`
```javascript
export async function getStaticPaths() {

    return {

        paths: [
            {
                params: {
                    rid: '1'
                }
            }
        ],
        fallback: true
    };
}

export async function getStaticProps(context) {

    console.log(context)

    return {
        props: {
            context
        }
    };
}
```

I would like to pass a query similar to the one mentioned in the documentation, I don't know if I'm doing something wrong in my code, or if I have to change the file name by two or more possible "variables" for example [rid,foo] or some other detail in getStaticPaths.

What will be the right way?

**Comment by jamesmosier** [comment]
`getStaticPaths` and `getStaticProps` do not work with query string parameters. Therefore it will only work with route parameters.

[Docs](https://nextjs.org/docs/basic-features/data-fetching#only-runs-at-build-time)

**Reply by ghost** [reply]
If I want to get a more complex parameter like the one I mentioned

So what should I use? getInitialProps?

I still don't understand the difference and uses of getStaticPaths, getStaticProps and getInitialProps  

:(
